expression/util.go

```diff
@@ -122,6 +124,35 @@ func ExtractColumns(expr Expression) []*Column {
 	return extractColumns(result, expr, nil)
 }
 
+// ExtractAllColumnsFromExpressionsInUsedSlices is the same as ExtractColumns. but it can reuse the memory.
+func ExtractAllColumnsFromExpressionsInUsedSlices(reuse []*Column, exprs ...Expression) []*Column {
+	if len(exprs) == 0 {
+		return nil
+	}
+	for _, expr := range exprs {
```

Comment:
cols = cols[:0]

Reply:
how about move to here, after all, your func name says it will reuse...

Reply:
Updated